cmd/soroban-cli/src/commands/config/mod.rs

```diff
@@ -49,6 +60,40 @@ impl Args {
         Ok(key.key_pair(self.hd_path)?)
     }
 
+    pub async fn sign_with_local_key(&self, tx: Transaction) -> Result<TransactionEnvelope, Error> {
+        self.sign(tx).await
+    }
+
+    pub async fn sign(&self, mut tx: Transaction) -> Result<TransactionEnvelope, Error> {
+        let key = self.key_pair()?;
+        let verifying_key = key.verifying_key().to_bytes();
+        let account = Strkey::PublicKeyEd25519(stellar_strkey::ed25519::PublicKey(verifying_key));
+        let network = self.get_network()?;
+        let client = Client::new(&network.rpc_url)?;
+        tx.seq_num = SequenceNumber(client.get_account(&account.to_string()).await?.seq_num.0 + 1);
+        tx.source_account = MuxedAccount::Ed25519(Uint256(verifying_key));
```

Comment:
Why do we overwrite the seq num and source account here during signing? These should have been set earlier in all cases. Signing should just involve attaching a signature of the tx.

Reply:
The idea here was that this uses the signing, which does just that, however, all all will need an accurate seq_num and given that the transaction could be out of date, I found it best to set it right before signing so that we won't get an error when sending a transaction with a bad seq_num. So this was more for the `tx sign`.

The source_account should be removed, especially because we should add a `sign_with` argument to allow for signing with a different key than the source account.

Ideally we would have a transaction builder *cough cough* as the logic of fetch and setting the seq_num is repeated over and over.

Reply:
I don't think we should modify the tx during the signing step. Signing in other tools is only a modification of the envelope, not the tx itself, which importantly means the tx hash is consistent, and the CLI magically modifying the seq would be surprising because it's not a different tx.

Agreed, the source account shouldn't be modified either for the reasons you mentioned.

Reply:
```suggestion
```